python/amici/sbml_import.py

```diff
@@ -884,6 +937,98 @@ def _process_time(self) -> None:
+            if parameter_target in self.symbols[SymbolId.SPECIES]:
+                continue
+            if parameter_target in self.parameter_assignment_rules:
+                raise SBMLException(
+                    'AMICI does not currently support models with SBML events'
+                    'that affect parameters that are also the target of '
+                    'assignment rules.'
+                )
+            parameter_def = None
+            for symbol_id in {SymbolId.PARAMETER, SymbolId.FIXED_PARAMETER}:
+                if parameter_target in self.symbols[symbol_id]:
+                    # `parameter_target` should only exist in one of the
+                    # `symbol_id` dictionaries.
+                    if parameter_def is not None:
+                        raise KeyError(
+                            'Unexpected error. The parameter target of an '
+                            'event assignment was processed twice.'
+                        )
+                    parameter_def = \
+                        self.symbols[symbol_id].pop(parameter_target)
+            if parameter_def is None:
+                raise KeyError(
+                    'Unexpected error. The parameter target of an event '
+                    'assignment could not be found.'
+                )
+            # Fixed parameters are added as species such that they can be
+            # targets of events.
+            self.symbols[SymbolId.SPECIES][parameter_target] = {
+                'name': parameter_def['name'],
+                'init': sp.Float(parameter_def['value']),
+                #'compartment': None,  # can ignore for amounts
+                'constant': False,
+                'amount': True,
+                #'conversion_factor': 1.0,  # probably can be ignored
+                'index': len(self.symbols[SymbolId.SPECIES]),
+                'dt': sp.Float(0),
+            }
+
+    @log_execution_time('processing SBML events', logger)
+    def _process_events(self) -> None:
+        """Process SBML events."""
+        events = self.sbml.getListOfEvents()
+        self._convert_event_assignment_parameter_targets_to_species()
+
+        for ievent, event in enumerate(events):
+            # get the event id (which is optional unfortunately)
+            event_id = event.getId()
+            if event_id is None:
+                event_id = f'event_{ievent}'
+            event_sym = sp.Symbol(event_id)
+
+            # get and parse the trigger function
+            trigger_sbml = event.getTrigger()
+            trigger_sym = self._sympy_from_sbml_math(trigger_sbml)
+            trigger = _parse_event_trigger(trigger_sym)
+
+            # Currently, all event assignment targets must exist in
+            # self.symbols[SymbolId.SPECIES]
+            state_vector = list(self.symbols[SymbolId.SPECIES].keys())
+
+            # parse the boluses / event assignments
+            bolus = [sp.Symbol('0') for _ in state_vector]
```

Comment:
Intended to be a symbol named `0`, instead of a value of `sp.Float(0.0)`?

Reply:
It at least works in the current version, but agree that the latter is at least more intuitive. Will change.

Reply:
It should work, because the symbol is just printed as `0` to the cpp files where it is interpreted as a number, but any `is_zero` will probably not yield the expected result...